File: python/LeetCode_121_BestTimetoBuyandSellStock.py
def maxProfit(prices) -> int:
    start, result = prices[0], 0
     
    for i in range(1, len(prices)):
        if (start > prices[i]):
            start = prices[i]
        elif (prices[i] - start > result):
            result = prices[i] - start 
    return result


print(maxProfit([2, 4, 1]))

File: c/LeetCode_121_BestTimetoBuyandSellStock.c
#include <stdio.h>
#include <stdlib.h>


int maxProfit(int* prices, int pricesSize) {
    if (pricesSize == 0) {
        return 0;
    }

    int start = prices[0];
    int result = 0;

    for (int i = 1; i < pricesSize; ++i) {
        if (start > prices[i]) {
            start = prices[i];
        } else if (prices[i] - start > result) {
            result = prices[i] - start;
        }
    }

    return result;
}


int main() {
    int pr[] = {2,4,1};
    int *p = pr;
    printf("%d", maxProfit(p, 3));

    return 0;
}